packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -290,6 +296,13 @@ export abstract class FunctionBase extends Resource implements IFunction {
     source.bind(this);
   }
 
+  public setAsyncInvokeConfig(id: string, options: EventInvokeConfigOptions): void {
```

Comment:
not sure I'm following here. Versions and aliases are of type `IFunction`, how do you want to set the qualifier when this method is called on an imported version or imported alias?

Reply:
Oops, my words got a bit garbled there. Let me try again.

I'm trying to address two things -

When a user calls `importedVersion.setAsyncInvokeConfig({ ... });`, they should not have to pass in a qualifier; it must be computed automatically (from the version ARN). The same goes for alias. 

At the same time, I'd rather not export a class named `BaseEventInvokeConfigOptions`. 
-- Classes with such names don't make it clear to the user (consumer or extender) what it represents, what it should contain and not contain, and when it should be used. Typically (there are times to exception), exported classes should make it clear from their name on either what they represent or when they must be used. --
So, in this case, we need it have a better name; or even better is to drop the need for such a class.

As for how this would be implemented, I believe there are several ways of doing this. A couple of ways that come to mind without deep thought - make `setAsyncInvokeConfig()` an abstract method on the base class which each of its subclass (including the `Import` classes) implement; or there could be a way to retrieving the qualifier from an instance of `QualifiedFunctionBase`. 
The best choice here would be based on how the code turns out to look.

Hope I was clearer this time.

Reply:
Renamed `setAsyncInvokeConfig` to `configureAsyncInvoke` since I removed the `id` parameter and `jsii` restricts from calling methods with one param `setXxx()`.

Added a `qualifier` property on `QualifiedFunctionBase` which allows to implement `configureAsyncInvoke()` on the abstract class.

We are now protected from multiple `AWS::Lambda::EventInvokeConfig` by the Construct id.